tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -109,8 +109,31 @@ const INT_DIV_VEC4 = `
   return vec4<f32>(resultTemp);
   `;
 
-const NOT_EQUAL = 'return f32(a != b);';
-const NOT_EQUAL_VEC4 = 'return vec4<f32>(a != b);';
+const NOT_EQUAL = `
```

Comment:
It seems that [webgl also doesn't check `isnan`](https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgl/src/kernels/NotEqual.ts#L23). Can you verify whether webgl has the similar issue on Linux?